### ISSUE TITLE
.pnpmfile.cjs does not support afterallresolved hooks

### ISSUE THREAD

**Issue by yangdan8** [pnpm/pnpm#4388]
### pnpm version:
6.32.1

### Code to reproduce the issue:

```cjs
// .pnpmfile.cjs

function readPackage(pkg, context) {
  console.log(1122, pkg.name);
  return pkg;
}

function afterAllResolved(lockfile, context) {
  console.log(7778888);
  return lockfile;
}

module.exports = {
  hooks: {
    readPackage,
    afterAllResolved,
  },
};
```

### Expected behavior:
log "1122" and "777888"

### Actual behavior:
log "1122" only

### Additional information:

 - `node -v` prints: 16.13.1
 - Windows, macOS, or Linux?: windows 11

**Comment by Cuxs** [comment]
Got the same issue here, is there any workaround?

**Comment by valendres** [comment]
I'm encountering this issue as well, but with a different version of pnpm. For me, if I use node `16.13.1` or node `18.12.1` with either pnpm `6.32.1` or  `6.35.1`, the `afterAllResolved` hook fires as expected. If I try to use pnpm `7.x` (either `7.0.0` or `7.25.1`), I encounter the same error as @yangdan8. It doesn't seem to matter what version of node I use. As for why I'm not able to reproduce this issue with the same versions, I suspect it may be because I'm running macOS instead of windows. 

TLDR, on macOS:
* **pnpm@6.x:** the `afterAllResolved` fires as expected
* **pnpm@7.x:** the `afterAllResolved` does not work.

**Comment by WhiteKiwi** [comment]
Got the same issue here, is there any workaround?

**Comment by WhiteKiwi** [comment]
node and pnpm version here 🙇 
<img width="178" alt="image" src="https://github.com/pnpm/pnpm/assets/32415680/3fe508af-abc6-4029-828e-09e42e3f2c67">

**Comment by neongreen** [comment]
I think this is supported, you're just not seeing the log because it gets overwritten.

Add `require('process').exit(1)` after your log and it will appear.

**Comment by linlinyang** [comment]
This bug has not fixed? I got the same problem.

**Comment by lake2** [comment]
same issue +1.
I guess `afterAllResolved` will not be called if current package.json is exactly mached pnpm-lock.yaml, because pnpm consider `nothing change, skip installing`.

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1: an `afterAllResolved` hook in .pnpmfile.cjs runs.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
